django/utils/dateformat.py

```diff
@@ -30,11 +30,17 @@
 
 
 class Formatter(object):
+
+    def format_piece(self, piece):
```

Comment:
Agreed, but also a question. Is the `validate_piece` function necessary at all? 
I could just check and raise error above append line:
`pieces.append(force_text(getattr(self, piece)())) `